### ISSUE TITLE
[meta] Logo Suggestions 

### ISSUE THREAD

**Issue by martinrieder** [wireviz/WireViz#373]
Any software deserves to have logo, so I am starting this issue for sharing and discussing any suggestions. With Daniel Rojas @formatc1702 being the originator of this great tool, he deserves the right to make the final choice. We could still have a community vote (before the next major release?) to come up with the best candidates.

**Comment by kvid** [comment]
I suggest using e.g. SVG to enable high-quality scaling when needed. Maybe see how these projects store logos:
- https://github.com/simple-icons/simple-icons
- https://github.com/gilbarbara/logos

**Comment by martinrieder** [comment]
I agree that there are some good reasons for choosing SVG, though I do not see this as a limiting factor by itself. It only implies some limitations regarding simplicity and coloring though... I feel confident that I can vectorize them using Inkscape. Contributions in raster images are also welcome, yet even pen-and-paper format if anyone here feels creative! 

I must admit that I left the creativity part to an AI, which came up with some interesting suggestions. It seems to be legasthenic though, so some spelling correction must be done... 

AFAIK there is no LLM that was trained on vector images, so I used DALL-E3. Let me know if you want to see the prompts for these. Some are minimalist and easy to vectorize, but others a bit more fancy and colorful. Please cast your vote and I will try my best to vectorize.

**Comment by martinrieder** [comment]
One
![Wireviz1.png](https://github.com/wireviz/WireViz/assets/74277074/f68912ad-4836-482b-8330-99688d0b6709)

**Comment by martinrieder** [comment]
Two
![_23c15786-ddcc-4994-bb9b-105adb310977..png](https://github.com/wireviz/WireViz/assets/74277074/600d2e52-16c7-4349-a80e-1bd21cec74ae)

**Comment by martinrieder** [comment]
Three
![_bb41c139-c36c-4239-b2fd-64985d37c1aa.jpeg](https://github.com/wireviz/WireViz/assets/74277074/96bf797f-97ed-4d2f-8bfb-1e7ee7d24fe2)

**Comment by martinrieder** [comment]
Four
![_7f555751-bc54-414e-abfc-8035e868ce6f.jpeg](https://github.com/wireviz/WireViz/assets/74277074/d8288779-6416-403c-801c-4a16e1964a04)

**Comment by martinrieder** [comment]
Five
![_c9eafda3-55a9-41e5-9eba-5b4b3149fe4d.jpeg](https://github.com/wireviz/WireViz/assets/74277074/c720ae61-ba70-490f-8089-584409461197)

**Comment by martinrieder** [comment]
Six
![_1777b269-19f5-4dce-84a5-ee5abf756c12.jpeg](https://github.com/wireviz/WireViz/assets/74277074/d73526d0-366c-4273-b282-29e7a01c9ba5)

**Comment by martinrieder** [comment]
Seven
![_56490524-f991-4985-8546-445c3c5adab2.jpeg](https://github.com/wireviz/WireViz/assets/74277074/9450046e-0564-48ab-b8b0-a801daa90b42)

**Comment by martinrieder** [comment]
Eight
![_37b2ac5d-7ac6-41a8-bf4d-9f5d12096695.jpeg](https://github.com/wireviz/WireViz/assets/74277074/81e0b1f1-a10f-4833-9dac-e830f6ef0b5c)

**Comment by martinrieder** [comment]
Nine
![_22da009d-5217-4f27-8d12-d0946e2cfba4.jpeg](https://github.com/wireviz/WireViz/assets/74277074/b8b0e363-95d3-4ea2-b7cc-daa57b356184)

**Comment by martinrieder** [comment]
Eleven
![_f35d4f24-350b-4bb5-837f-04b025eff0e9.jpeg](https://github.com/wireviz/WireViz/assets/74277074/ecedf0c0-c03e-4562-96e9-ba0985130760)

**Comment by martinrieder** [comment]
Twelve
![_5f56c667-4641-4d14-a9bb-20af82f1b232.jpeg](https://github.com/wireviz/WireViz/assets/74277074/f12599e6-0b5f-4acd-abf1-9fbaaec06ee9)

**Comment by tobiasfalk** [comment]
They all look good  but I think that most of them are tomuch, crowded, complicated.
Something simpler would be much better, nine, goes in to the right direction.

Consider how they work in different sizes, like a github profile picture or a desktop icon.

**Comment by kvid** [comment]
I agree with most arguments by @tobiasfalk that a simple logo is better. It should IMHO be easily recognizable also when shown as small icons, e.g. in Windows. Of the 12 suggestions, I've seen so far, nine seems better, but it might be too similar to the USB logo.

@tobiasfalk  wrote:
> [...] Should the logo have a comandline/terminal representation and so on.

Are you thinking about an ASCII art version of a graphical logo?

**Comment by tobiasfalk** [comment]
> @tobiasfalk  wrote:
> > [...] Should the logo have a comandline/terminal representation and so on.
> 
> Are you thinking about an ASCII art version of a graphical logo?
> 

Yes

**Comment by martinrieder** [comment]
Thanks for the feedback. I understand your concerns. Please see the images above as an inspiration. None of them are meant to be final. We could also merge certain aspects into a combined version.

There's another one I left out because I thought that it might be suited as an icon for the YAML files. 
![Wireviz3.png](https://github.com/wireviz/WireViz/assets/74277074/62c53f11-1baf-4915-97b8-7f482f363c08)

That brings up the question: does the logo need to include the WireViz text?

**Comment by kvid** [comment]
@martinrieder wrote:
> [...] 
> That brings up the question: does the logo need to include the WireViz text?

IMHO - no, because that is unrealistic to recognize for small icon sizes. The name is too long.

**Comment by tobiasfalk** [comment]
@martinrieder this also looks like a good start, as said a more simpler Version could be more effective in recognition and so, especially when small.

As for the the WireViz in the logo, I would say maybe not  the whole name but a W and a V would be a good idea.

These two arrangements come to my minde:

V
W

And

W
V

**Comment by tobiasfalk** [comment]
![Screenshot_20240602-201441_Keep Notes.jpg](https://github.com/wireviz/WireViz/assets/56540480/838a77c1-95fd-4d78-8ed6-8f53245ab6f5)

![Screenshot_20240602-201908_Keep Notes.jpg](https://github.com/wireviz/WireViz/assets/56540480/7889eeb4-6787-45ca-8b5e-5246208cb7e3)

**Comment by tobiasfalk** [comment]
https://youtu.be/YPoLD_5FQ38?si=7bf4MCppmUOhpJVl
Sorry😁🤦‍♂️

**Comment by kvid** [comment]
If you search for WV logo, you'll find West Virginia and a few others that we should be careful about getting too similar to.

**Comment by tobiasfalk** [comment]
WZ could also work, since the Z is more prominent when saying Viz

**Comment by martinrieder** [comment]
@kvid wrote:
> I suggest using e.g. SVG to enable high-quality scaling when needed. Maybe see how these projects store logos:
> - https://github.com/simple-icons/simple-icons
> - https://github.com/gilbarbara/logos

I am well aware of the trend to have minimalist logos, optionally spelling out the name separately. Knowing what stands behind them is often hard and comes close to learning Egyptian hieroglyphics. In that respect, I would say that we should not incorporate any letters into the logos. It needs to show some connectors and wires though.

**Comment by tobiasfalk** [comment]
What about a Plan where a wire comes out of?

**Comment by martinrieder** [comment]
> What about a Plan where a wire comes out of?

![_aee788b0-6bcb-430e-9f44-34350343a556.jpeg](https://github.com/wireviz/WireViz/assets/74277074/7404a27e-c5dd-4c4d-81aa-7da2816e8e34)

**Comment by martinrieder** [comment]
> > What about a Plan where a wire comes out of?
> 
> ![_aee788b0-6bcb-430e-9f44-34350343a556.jpeg](https://github.com/wireviz/WireViz/assets/74277074/7404a27e-c5dd-4c4d-81aa-7da2816e8e34)
> 
> 

This one I had already saved. Any ideas for simplification?

**Comment by tobiasfalk** [comment]
Maybe only one paiper(or half of it) and one wire with connector.
![Screenshot_20240602-225242_Keep Notes.jpg](https://github.com/wireviz/WireViz/assets/56540480/b989ec9d-b4bb-4aa2-845e-f737beb6ba1d)

**Comment by martinrieder** [comment]
> Maybe only one paiper(or half of it) and one wire with connector.

IMHO a wiring harness would always contain multiple wires and connectors... 

Thinking opposite, we could also show a connector _going into_ the schematics sheet.

**Comment by tobiasfalk** [comment]
> Thinking opposite, we could also show a connector _going into_ the schematics sheet. 

Yes sounds good, but isn't WireViz used to design harneses before creating them, and this may look the other way around.

Since designing is often connected with pens, why not a pen or pencil and wires somewhere/how.

**Comment by martinrieder** [comment]
> > Thinking opposite, we could also show a connector _going into_ the schematics sheet. 
> 
> Yes sounds good, but isn't WireViz used to design harneses before creating them, and this may look the other way around.
> 
> Since designing is often connected with pens, why not a pen or pencil and wires somewhere/how.
> 

Funny, because that is one thing that I also got as a suggestion. The prompt was: "Create a logo with a minimalistic style in rectangular shape, featuring connectors in a wiring harness surrounded by a frame."

![_a0534f63-c74d-4c41-a859-8041b5a169e1.jpeg](https://github.com/wireviz/WireViz/assets/74277074/bcf7d943-fabf-44db-9d33-654d324324b9)

**Comment by martinrieder** [comment]
Here's a connector going outwards:
![_f00064b3-766c-477d-b15d-672eef662198.jpeg](https://github.com/wireviz/WireViz/assets/74277074/406c7db5-1e29-4dfd-99c0-f555e3cdf77a)

**Comment by tobiasfalk** [comment]
> Here's a connector going outwards:
> ![_f00064b3-766c-477d-b15d-672eef662198.jpeg](https://github.com/wireviz/WireViz/assets/74277074/406c7db5-1e29-4dfd-99c0-f555e3cdf77a)
> 
> 

Maybe just the circle

**Comment by martinrieder** [comment]
![_19e2dba7-af25-40cd-b26c-872a3f7ffc4f.jpeg](https://github.com/wireviz/WireViz/assets/74277074/3f959f61-1273-4e81-8dda-bbb45522b1fa)

Is this one is still too crowded?

**Comment by tobiasfalk** [comment]
And more wires than traces

**Comment by martinrieder** [comment]
![_2106be87-5680-4372-b7db-a5f3ea2deedc.jpeg](https://github.com/wireviz/WireViz/assets/74277074/e8115096-4c4e-49e2-accd-eb9c2467ba6f)

My last one for today. Just some wires and really simple connectors.

**Comment by martinrieder** [comment]
> And more wires than traces 

I can remove the circles resembling vias in a PCB trace of course.

**Comment by obelisk79** [comment]
Greetings, I recently led the effort to replace the logo for FreeCAD. As part of that I spend some time studying logo/design theory. I would note that a logo should be visually simple, unique and easily identifiable from other products. Using letters such as WV aren't really that important as long as the visuals are clean, scale properly and don't rely on color to be effective.

It seems like there's a lot of AI generated images in this discussion. I would caution against this approach. However, of the images shown, No. 9 seems to be the closest to a decent concept.

**Comment by martinrieder** [comment]
@obelisk79 thanks for your feedback. I noticed that www.freecad.org shows a different logo than wiki.freecad.org. Is there a reason for this?  Which one do you prefer?

The AI content is only intended as a source of inspiration. None of them are meant to be the final logo. Some editing will be required anyways.

**Comment by tobiasfalk** [comment]
> I noticed that www.freecad.org shows a different logo than wiki.freecad.org. Is there a reason for this?
This has most likely to do with the fact that the new Logo was finalised just some weeks ago and that not all is moved.

**Comment by obelisk79** [comment]
This is the correct new logo for FreeCAD. The migration has not been instant but most instances have been updated:
![image](https://github.com/wireviz/WireViz/assets/17993710/f8660e9d-b303-4e77-8ff3-3be3ca51e73d)

**Comment by obelisk79** [comment]
I did a little doodling based on image #9 
![image](https://github.com/wireviz/WireViz/assets/17993710/c97b6fd4-c79f-4c0c
![drawing-1](https://github.com/wireviz/WireViz/assets/17993710/0b5ea562-9e03-4579-95a4-5589d2c5386b)
-a2ea-7418114f0b63)

**Comment by martinrieder** [comment]
@obelisk79 wrote:
> I did a little doodling based on image #9 

Thanks for that idea. It looks nice, though I think that it is missing some important characteristics of a wiring harness:

- Diversity of connectors
- Spliced connections
- Bundling of wires

Therefore, I have created something similar, though not as sleek as yours:
![image1.png](https://github.com/wireviz/WireViz/assets/74277074/30bcd779-0186-4c37-8ba1-a39fab7cec01)

**Comment by obelisk79** [comment]
You could add those elements in, I actually had splice dots and removed them, a full representation isn't necessary. It's abstract and simple. Either way, I attached the vector file above. I created it in inkscape.

**Comment by martinrieder** [comment]
> [.. .] I thought it should be left-right oriented like a real WireViz diagram. Spinning a bit further on that thought, made me try out how I could use WireViz to create a logo suggestion. [...] My goal is to capture the essence of WireViz in a simple logo that is easy to recognize and easy to distinguish from other logos.

I love the idea. It provides a proof of the capabilities that WireViz has. Thanks for sharing the YAML code on this piece of "artwork".

> The small 16x16px version is very blurry - how can that be avoided? Is it better to reduce the number of connectors and wires from 3 to 2 and scale up pen thickness, or does it have to be hand drawn?

Why does it need to be that small if the trend is moving in another direction? Could we have different levels of detail for different resolutions? 

My humble idea that I have not tried yet:
- 4 connectors to the left, labeled with the letters W, I, R, E
- 2 wires only to have them crossed and spliced
- 3 connectors to the right, containing the letters V, I, Z

Nice Idea. There is an online tool called [SketchViz](https://www.sketchviz.com) that will generate a hand-drawn version of your graph.

My personal favorite is actually number two, of which I made a simplified version. It still needs some polishing though:

https://github.com/wireviz/WireViz/issues/373#issuecomment-2143866846
> EDIT two simplified
![image4.png](https://github.com/wireviz/WireViz/assets/74277074/8cc6edda-d456-41e8-8db6-cfe4ce54b96c)

**Comment by kvid** [comment]
@martinrieder wrote:
> [...]
> > The small 16x16px version is very blurry - how can that be avoided?
> 
> Why does it need to be that small if the trend is moving in another direction? Could we have different levels of detail for different resolutions?

16x16px seems to be the smallest icon size used at the relevant OS platforms:
https://blog.icons8.com/articles/choosing-the-right-size-and-format-for-icons/#h-desktop

**Comment by tobiasfalk** [comment]
@martinrieder yes looks good, could you put it on a white background?

**Comment by obelisk79** [comment]
Looks decent. As far as the properties stated in #373, I'd disregard them. In the world of graphic/logo design we aren't necessarily conveying purpose as much as we want a unique and easily identifiable design loosely associated with the purpose of the brand/product. This checks those boxes fairly well.

**Comment by formatc1702** [comment]
Hi,
thanks for getting this discussion going. It's true that it would be nice to give the project a bit of a visual identity.

Some rough points from my side. Very personal opinions obviously, and up for discussion:

- All the AI-generated logos are very cluttered and lack some soul (i.e. they clearly look AI made), so I like that the conversation is going towards simplification
- Having letters/text in the logo can "cheapen" it if not done carefully
- Having the logo resemble WireViz output could be clever, but might risk making it too bland. Depends on the implementation.

So far, I see the most potential in:

- <img width="100" alt="Screenshot 2024-07-05 at 13 23 25" src="https://github.com/wireviz/WireViz/assets/2447251/380b710e-1f9f-41bc-ba6a-70e52fc3aa92"> by @obelisk79
  With a bit of imagination, one can almost see a W and a V. It's a VW more than a WV but subtle and far enough from Volkswagen. The connectors look a bit like Lego figures wearing tophats 🎩 which is not necessarily bad 😄 Regarding the splice dots, to me that is more reminiscent of a PCB-level schematic, and (currently), dots are not used as splices in WireViz, so I don't miss them.
- <img width="100" alt="Screenshot 2024-07-05 at 13 23 07" src="https://github.com/wireviz/WireViz/assets/2447251/3b3cedee-cc35-41ca-a36f-367bdfabdd0b"> by @martinrieder 
  More reminiscent of WireViz output. It evokes a letter E; perhaps rotating it can help evoke a W. Generally, if done well, a W without V could be iconic enough.

---
  
For what it's worth, here is my own attempt, quickly sketched out and terribly colored in.

Full version:
<img width="300" 
src="https://github.com/wireviz/WireViz/assets/2447251/df592b44-c7cd-4428-80fe-3fd176e48d1b"> <img width="300" 
src="https://github.com/wireviz/WireViz/assets/2447251/b6fd0cbb-44d3-4021-b1d3-62844d94f3db">

Simplified icon version (64x64, 32x32, 16x16, downscaled in browser, looks good on retina)
<img width="64" 
src="https://github.com/wireviz/WireViz/assets/2447251/88afd629-101d-42b4-accc-4f9171b783af"> <img width="64" 
src="https://github.com/wireviz/WireViz/assets/2447251/136ef538-5665-4ac2-a4ac-63c34d85499a"> <img width="32" 
src="https://github.com/wireviz/WireViz/assets/2447251/88afd629-101d-42b4-accc-4f9171b783af"> <img width="32" 
src="https://github.com/wireviz/WireViz/assets/2447251/136ef538-5665-4ac2-a4ac-63c34d85499a"> <img width="16" 
src="https://github.com/wireviz/WireViz/assets/2447251/88afd629-101d-42b4-accc-4f9171b783af"> <img width="16" 
src="https://github.com/wireviz/WireViz/assets/2447251/136ef538-5665-4ac2-a4ac-63c34d85499a">

Icon version (true 16x16 px, retina screens might apply some blurry upscaling)
<img width="16" 
src="https://github.com/wireviz/WireViz/assets/2447251/bff39f63-43ef-4a39-b4ae-548f235f10aa"> <img width="16" 
src="https://github.com/wireviz/WireViz/assets/2447251/c567239f-3603-43fd-959f-0cb5f6331d02">

- Simple basic structure: One connector on one side, a curl, a splice, two connectors/wires on the other end
- The W is quite obvious, the V is a bit more subtle
- Should work in color and B/W
- Colors were chosen quite arbitrarily
- The line thickness on the large version could probably be optimized
- 16x16 retina version is readable, true 16x16 would need some pixel tuning
- I like the proportions of the icon version better, the V on the full version is quite slanted.
- Would need to be paired with an appropriate font (also curly for cohesion, or more angular as a contrast?) when combining logo with "WireViz" spelled out next to it.

If anybody wants to try their hand at digitizing/fine-tuning it, go ahead 🚀  If I find the time, I might give it a go as well.
